File: modules/swagger-jaxrs/src/main/scala/com/wordnik/swagger/jaxrs/config/FilterFactory.scala
package com.wordnik.swagger.jaxrs.config

import com.wordnik.swagger.core.filter._
import com.wordnik.swagger.model._

object FilterFactory {
  var filter: SwaggerSpecFilter = new DefaultSpecFilter
}

class DefaultSpecFilter extends SwaggerSpecFilter {
  def isOperationAllowed(operation: Operation, api: ApiDescription, params: Map[String, List[String]], cookies: Map[String, String], headers: Map[String, List[String]]): Boolean = true
  def isParamAllowed(parameter: Parameter, operation: Operation, api: ApiDescription, params: Map[String, List[String]], cookies: Map[String, String], headers: Map[String, List[String]]): Boolean = true
}

File: modules/swagger-core/src/main/java/io/swagger/config/FilterFactory.java
package io.swagger.config;

import io.swagger.core.filter.OpenAPISpecFilter;

public class FilterFactory {
    protected static OpenAPISpecFilter FILTER = null;

    public static OpenAPISpecFilter getFilter() {
        return FILTER;
    }

    public static void setFilter(OpenAPISpecFilter filter) {
        FILTER = filter;
    }
}